arch/riscv/Kconfig

```
@@ -46,6 +46,37 @@ config INCLUDE_RESET_VECTOR
 	  Include the reset vector stub, which initializes the stack and
 	  prepares for running C code.
 
+config RISCV_SOC_ISR_STACKING
+	bool "SOC-based ISR stacking / unstacking"
+	help
+	  Enable low-level SOC-specific hardware stacking / unstacking
+	  operations during ISR.
+
+	  Some SOCs implement a mechanism for which, on interrupt handling,
+	  part of the context is automatically saved by the hardware on the
+	  stack according to a custom ESF format. The same part of the context
+	  is automatically restored by hardware on mret.
+
+	  Enabling this option requires that the SoC provides a soc_stacking.h
+	  header which defines the following:
+
+	  - SOC_ISR_SW_STACKING: macro guarded by _ASMLANGUAGE called by the
+	    IRQ wrapper assembly code on ISR entry to save in the ESF the
+	    remaining part of the context not pushed already on the stack by
+	    the hardware.
+
+	  - SOC_ISR_SW_UNSTACKING: macro guarded by _ASMLANGUAGE called by the
+	    IRQ wrapper assembly code on ISR exit to restore the part of the
+	    context from the ESF that won't be restored by hardware on mret.
+
+	  - SOC_ISR_STACKING_ESF: structure declaration for the ESF guarded by
```

We tipically use define/declare in macros that do so:

```suggestion
	  - SOC_ISR_STACKING_ESF_DECLARE: structure declaration for the ESF guarded by
```

---

fixed